Framework/PythonInterface/test/python/mantid/kernel/ConfigServiceTest.py

```diff
@@ -92,10 +93,74 @@ def test_setting_paths_via_single_string(self):
         self.assertTrue('tmp_2' in paths[1])
         self._clean_up_test_areas()
-            
+
     def test_setting_log_channel_levels(self):
         testhelpers.assertRaisesNothing(self, config.setLogLevel, 4, True)
-    
+
+    def test_properties_documented(self):
+        # location of the rst file relative to this file this will break if either moves
+        doc_filename = os.path.split(inspect.getfile(self.__class__))[0]
+        doc_filename = os.path.join(doc_filename, '../../../../../../docs/source/concepts/PropertiesFile.rst')
+        doc_filename = os.path.abspath(doc_filename)
+
+        # read in the user documentation
+        print ('Parsing', doc_filename)
+        documented_keys = []
+        with open(doc_filename) as handle:
+            text = handle.read()
+
+        # these will be ignored - the list should get shorter over time
+        hidden_prefixes = ['CheckMantidVersion.DownloadURL',  # shouldn't be changed by users
+                           'CheckMantidVersion.GitHubReleaseURL', # shouldn't be changed by users
+                           'UpdateInstrumentDefinitions.URL', # shouldn't be changed by users
+                           'docs.html.root', # shouldn't be changed by users
+                           'usagereports.rooturl', # shouldn't be changed by users
+
+                           ########## TODO should these be documented?
+                           'ISISDAE.Timeout',
+                           'PeakColumn.hklPrec',
+                           'catalog.timeout.value',
+                           'cluster.submission',
+                           'colormaps.directory',
+                           'curvefitting.defaultPeak', 'curvefitting.findPeaksFWHM', 'curvefitting.findPeaksTolerance',
+                           'errorreports.rooturl',
+                           'filefinder.casesensitive',
+                           'graph1d.autodistribution',
+                           'groupingFiles.directory',
+                           'icatDownload.directory', 'icatDownload.mountPoint',
+                           'instrument.view.geometry',
+                           'interfaces.categories.hidden',
+                           'loading.multifile', 'loading.multifilelimit',
+                           'logging.channels.consoleChannel.class', 'logging.channels.consoleChannel.formatter', 'logging.formatters.f1.class', 'logging.formatters.f1.pattern', 'logging.formatters.f1.times', 'logging.loggers.root.channel.channel1', 'logging.loggers.root.channel.class',
+                           'mantidqt.python_interfaces', 'mantidqt.python_interfaces_directory',
+                           'maskFiles.directory',
+                           'openclKernelFiles.directory',
+                           'paraview.ignore', 'paraview.path', 'paraview.pythonpaths', 'pvplugins.directory',
+                           'python.plugins.directories',
+                           'pythonalgorithms.refresh.allowed',
```

Comment:
I don't think we use this key any more so we could remove it? I wonder if any others are dead?